clients/src/main/java/org/apache/kafka/clients/admin/internals/ListConsumerGroupOffsetsHandler.java

```diff
@@ -110,41 +112,97 @@ public String apiName() {
                         groupOffsetsListing.put(topicPartition, new OffsetAndMetadata(offset, leaderEpoch, metadata));
                     }
                 } else {
-                    log.warn("Skipping return offset for {} due to error {}.", topicPartition, error);
+                    handlePartitionError(groupId, topicPartition, error, groupsToUnmap, groupsToRetry);
                 }
             }
             completed.put(groupId, groupOffsetsListing);
         }
-        return new ApiResult<>(completed, failed, unmapped);
+
+        if (groupsToUnmap.isEmpty() && groupsToRetry.isEmpty()) {
+            return new ApiResult<>(
+                completed,
```

Comment:
We could get rid of `completed` and use `Collections.singletonMap(groupId, groupOffsetsListing)`, no?

Reply:
No, we can't do that because the `completed` here could be `empty map`. If we put `Collections.singletonMap(groupId, groupOffsetsListing)`, it'll always not empty. Thanks.

Reply:
@showuon I think that there is a case that we don't handle correctly.

Imagine that `GROUP_AUTHORIZATION_FAILED` is returned as a partition error. In this case, we ignore it in `handlePartitionError` and therefore don't add the failed group to `failed`. I think that we should also handle all the group level errors in `handlePartitionError`.

The second thing is that if there is a group failure, we should not add the group to `completed` at L131. Otherwise, this will complete the group future with an empty list.

Could you check this out and add a test for it?

Reply:
Good suggestion! Will do it tomorrow (my time). Thanks.